magma/type_utils.py

```diff
@@ -0,0 +1,110 @@
+import dataclasses
```

Comment:
This file and `value_utils.py` are some common useful utilities that might be useful in other places as well. Let me know any feedback you have that might make these more useful.